airflow/providers/google/common/operators/base_google.py

```diff
@@ -0,0 +1,34 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a Google API base operator."""
+from __future__ import annotations
+
+from google.api_core.gapic_v1.method import DEFAULT
+
+from airflow.models import BaseOperator
+
+
+class GoogleBaseOperator(BaseOperator):
+    """
+    Abstract base class that takes care of common specifics of the operators built
+    on top of Google API client libraries.
+    """
```

Comment:
Our goal is to split Google provider to smaller providers https://github.com/apache/airflow/issues/15933
This class seems to be GCP specific so lets keep it in the `cloud` directory and not in `common` as this directory will not stay after the split.

Possibly also rename the class to`GcpBaseOperator` ?

Reply:
Good idea, especially since AFAIU only Cloud providers are affected. GCP is a bit of a deprecated term, but "Google Cloud Base" should do that.